### ISSUE TITLE
Forward port for ingress controller label issues

### ISSUE THREAD

**Issue by kinarashah** [rancher/rancher#30208]
Forward port fix in both 2.5 and master for the following issues:  
https://github.com/rancher/rancher/issues/30045 
https://github.com/rancher/rancher/issues/30044

**Comment by sowmyav27** [comment]
On 2.5-head - commit id: `b2953cbc`

- Create cluster with Project Network Isolation Enabled 1 etcd/control and 2 worker nodes.
- Pods of `nginx-ingress-controller` in system project have podName labels

```
labels:
    app: ingress-nginx
    controller-revision-hash: 6f4d9bb968
    field.cattle.io/podName: nginx-ingress-controller-fgcm4
    pod-template-generation: "1"
```
- Ingress is stuck in Initializing state

On master-head, the ingress do not get all the worker nodes' ip addresses.

**Comment by sowmyav27** [comment]
### Fresh install usecase - On 2.5-head - commit id: `baf4867c`

- Create cluster with Project Network Isolation Enabled 1 etcd/control and 2 worker nodes.
- Pods of `nginx-ingress-controller` in system project do not have podName labels.
- In Debug mode, redeploy `nginx-ingress-controller` and confirm the pods don't get updated without cause. `podHandler: addLabelIfHostPortsPresent: deleting podNameFieldLabel` is NOT seen in logs.
- Deploy user ingress, verified the status.LoadBalancer shows IPs of all the worker nodes - 2 worker nodes.
```
status:
  loadBalancer:
    ingress:
    - ip: <ip-1>
    - ip: <ip-2>
```
---
### Upgrade from 2.5.2 to 2.5-head
On 2.5.2
- Deploy a cluster - 2 worker nodes, 1 etcd/control plane node
- Enable cluster level monitoring
- Deploy a workload and an ingress pointing to the workload.
- User ingress has only one worker node's ip address in status.LoadBalancer
```
status:
  loadBalancer:
    ingress:
    - ip: <wk01>
```
- ingress controller pods in the system project have podName labels
- Create network policies using [doc](https://kubernetes.io/docs/concepts/services-networking/network-policies/#networkpolicy-resource)
- Upgrade rancher to 2.5-head - blocked because of issue - https://github.com/rancher/rancher/issues/30434

**Comment by kinarashah** [comment]
Available to test in `v2.5-head`

**Comment by sowmyav27** [comment]
Upgrade from 2.5.2 to 2.5.4-rc4

On 2.5.2
- Deploy a cluster - 4 worker nodes, 1 etcd/control plane node
- Enable monitoring on the cluster
- Deploy a workload and an ingress pointing to the workload.
- User ingress has only all the worker node's ip address in status.LoadBalancer
- Ingress controller pods have podName labels - `field.cattle.io/podName: nginx-ingress-controller-2bzlh`
- Upgrade to 2.5.4-rc4
- Rancher logs show: `2020/12/12 02:57:07 [DEBUG] podHandler: addLabelIfHostPortsPresent: deleting podNameFieldLabel map[app:ingress-nginx controller-revision-hash:5dd9dd4479 field.cattle.io/podName:nginx-ingress-controller-gbch6 pod-template-generation:1] in ingress-nginx`
- User ingress is seen in `Active` state.
- pods of nginx-ingress-controller in system project don't have podName labels
- User ingress has all the worker nodes' ip address in status.LoadBalancer 
- Monitoring metrics -CPU usage decreased from 100% to 20%..
- Kubelet logs on the worker node do not show repeated Sync Loop updates/message
